src/libraries/System.Private.CoreLib/src/System/CharEnumerator.cs

```diff
@@ -35,22 +35,27 @@ public object Clone()
 
         public bool MoveNext()
         {
-            if (_index < (_str!.Length - 1))
+            int index = _index + 1;
+            string s = _str!;
+
+            if ((uint)index < (uint)s.Length)
             {
-                _index++;
-                _currentElement = _str[_index];
+                _currentElement = s[index];
```

Comment:
Is it worth caching the current element? `Current` is reading both `s` and `_index` and doing a boundcheck on it. It should be very cheap to just return `s[_index]` there.

Reply:
At least on my machine, changing it to not have _current ends up being a little bit slower than what was in the PR.  But we're talking a few percent, so I changed it anyway to opt for the simpler version.